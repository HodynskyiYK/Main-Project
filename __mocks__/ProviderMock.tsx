import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import { MovieModalContextProvider } from '../src/context/movie-modal-context'
import { Provider } from 'react-redux'
import { store } from '../src/store'
import { MovieDetailsContextProvider } from '../src/context/movie-details-context'

// @ts-ignore
export const ProvidersMock = ({ children }) => {
    return (
        <Provider store={store}>
            <MovieModalContextProvider>
                <MovieDetailsContextProvider>
                    <MemoryRouter>{children}</MemoryRouter>
                </MovieDetailsContextProvider>
            </MovieModalContextProvider>
        </Provider>
    )
}