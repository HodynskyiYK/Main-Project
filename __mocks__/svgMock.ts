// Mock Svg files
// https://github.com/gregberge/svgr/issues/83#issuecomment-437342861
const SvgMock = "icon-mock"
export const ReactComponent = SvgMock
export default SvgMock