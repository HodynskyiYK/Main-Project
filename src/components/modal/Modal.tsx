import React, { FC } from 'react'
import { IModal } from './modalTypes'
import styles from './Modal.module.scss'
import { useMovieModalContext } from '../../context/movie-modal-context'
import { CLOSE_ICON_23_23 } from '../../assets/svg'

export const Modal: FC<IModal> = ({children}) => {
    const {
        hideAddMovieModal,
        hideEditMovieModal,
        hideDeleteMovieModal
    } = useMovieModalContext()

    const hideModal = () => {
        hideAddMovieModal()
        hideEditMovieModal()
        hideDeleteMovieModal()
    }

    return (
        <div className={styles.modal}>
            <div className={styles.modalContent}>
                <button
                    className={styles.closeModal}
                    onClick={hideModal}
                >
                    {CLOSE_ICON_23_23}
                </button>
                {
                    children
                }
            </div>
        </div>
    )
}
