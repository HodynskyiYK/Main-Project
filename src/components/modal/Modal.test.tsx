import React from 'react'
import { Modal } from './Modal'
import { render, screen } from '@testing-library/react'
import { ProvidersMock } from '../../../__mocks__/ProviderMock'

describe('Modal tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<ProvidersMock>
            <Modal>
                <span>Test</span>
            </Modal>)
        </ProvidersMock>)
        expect(asFragment()).toMatchSnapshot()
    })

    test('close button in modal', () => {
        render(<ProvidersMock>
            <Modal>
                <span>Test</span>
            </Modal>)
        </ProvidersMock>)
        const btn = screen.getByTestId('close-btn')
        expect(btn).toBeInTheDocument()
    })
})