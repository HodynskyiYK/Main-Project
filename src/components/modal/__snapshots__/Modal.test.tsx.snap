// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal tests To match snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modalContent"
    >
      <button
        class="closeModal"
        data-testid="close-btn"
      />
      <span>
        Test
      </span>
    </div>
  </div>
  )
</DocumentFragment>
`;
